### ISSUE TITLE
Dilate script causes out of bounds

### ISSUE THREAD

**Issue by snail-with-tea** [guillaumechereau/goxel#352]
Goxel v0.14.0
Linux 6.7.1-zen

![Screenshot_20240205_224052](https://github.com/guillaumechereau/goxel/assets/99575791/eb089f3b-ccd8-4bd0-afad-5abd2724b36b)

### Steps to reproduce:
1. Draw anything on bounds
2. Use Scripts>Dilate

This should give similar results to image above

**Comment by snail-with-tea** [comment]
Just figured out that Extrude tool does OoB too:
![Screenshot_20240205_225232](https://github.com/guillaumechereau/goxel/assets/99575791/44dafa8c-d795-40c6-af97-1d2dba8b5113)

**Comment by guillaumechereau** [comment]
Yes I'll check that.  For now you can go in the layer and press the 'crop to image' button.
